source/gameengine/Ketsji/KX_FontObject.cpp

```diff
@@ -105,6 +106,32 @@ KX_FontObject::~KX_FontObject()
 	//it's handled in KX_Scene::NewRemoveObject
 }
 
+const MT_Vector2 KX_FontObject::GetTextDimensions()
+{
+	MT_Vector2 dimensions;
+
+	for (std::vector<STR_String>::iterator it = m_text.begin();
+		 it != m_text.end(); ++it)
+	{
+		float w = 0.0f, h = 0.0f;
+		STR_String text = (*it);
+
+		BLF_width_and_height(m_fontid, text.ReadPtr(), text.Length(), &w, &h);
+		dimensions.x() = std::max(dimensions.x(), w);
+		dimensions.y() += h + m_line_spacing;
+	}
+
+	// XXX: Quick hack to convert the size to BU
+	dimensions /= 10.0f;
+
+	// Scale the width and height by the object's scale
+	MT_Vector3 scale = NodeGetLocalScaling();
+	dimensions.x() *= scale.x();
```

Comment:
fabs(scale.x()) ? I don't know when the object has a negative scale on X axis for example it returns (-11.123, 2.12)
The dimensions shouldn't always be positive?